### PR TITLE
Add hint and clarity to project descriptions in creation form (IDSEQ-2162)

### PR DESCRIPTION
**Pull request by lvreynoso**
# Description

This change adds a toggleable infobox to the "Project Description" field on the project creation form. The infobox helps clarify what is expected in a project description. A more concise version is used as placeholder text for the text input field itself.

# Tests
Select "Upload" from the top right menu. Click "Create Project". Toggle the "More Info" button.

Screen Shot 2020-02-06 at 9 29 27 AM

### CODE REVIEW

File: app/assets/src/components/common/project_creation_form.scss [hunk 1]
```diff
@@ -5,6 +5,13 @@
   .label {
     @include font-header-s;
     margin-bottom: 5px;
+
+    .infoLink {
+      @include font-label-s;
+      margin-left: 10px;
```
Comment (j-x-han) [inline app/assets/src/components/common/project_creation_form.scss]: We have some spacing variables in `_elements.scss` that could be used here 😄 (`$space-default` for 10px)

**Comment by happyimadesignr** [comment]
Awesome Lucia!

I had a few comments =D:
* We wanted to decrease the text area size
* Also wanted to add a "Select Project" title to the project creation, to help make it stand out better.
* Box needs rounded corners
* Move the “More Info” button closer to “Project Description” like the mocks, use spacing variable if not already
* Placeholder copy is too close to the edge of the text area box, can you add more padding, use spacing variables if not already

Thanks!!!

**Comment by lvreynoso** [comment]
updated!

Screen Shot 2020-02-06 at 11 21 56 AM

File: app/assets/src/components/common/ProjectCreationForm.jsx [hunk 2]
```diff
@@ -60,7 +62,21 @@ class ProjectCreationForm extends React.Component {
     }
   };
 
+  toggleInfo = () => {
+    const { showInfo } = this.state;
+
+    logAnalyticsEvent("ProjectCreationForm_more-info-toggle_clicked", {
+      showInfo,
```
Comment (gregdingle) [inline app/assets/src/components/common/ProjectCreationForm.jsx]: nit: this might be misleading when `showInfo` `true` means close the box

Comment (lvreynoso) [reply app/assets/src/components/common/ProjectCreationForm.jsx]: Right, I should do this in a callback when setting the new state. Thanks for catching!

File: app/assets/src/components/common/project_creation_form.scss [hunk 3]
```diff
@@ -15,10 +23,12 @@
   .descriptionTextArea {
     @include font-body-xs;
     margin-top: 5px;
-    padding: 5px;
-    height: 150px;
+    // margin-left: -5px;
```
Comment (jshoe) [inline app/assets/src/components/common/project_creation_form.scss]: nit: remove L26

File: app/assets/src/components/views/SampleUploadFlow/UploadSampleStep.jsx [hunk 4]
```diff
@@ -698,6 +698,7 @@ class UploadSampleStep extends React.Component {
       >
         <div className={cs.flexContent}>
           <div className={cs.projectSelect}>
+            <header>Select Project</header>
```
Comment (jshoe) [inline app/assets/src/components/views/SampleUploadFlow/UploadSampleStep.jsx]: can you change this to a `div` and `cs.header`? For consistency

**Pull request merged**